internal/cloudapi/v2/handler.go

```diff
@@ -627,10 +640,28 @@ func (h *apiHandlers) getComposeStatusImpl(ctx echo.Context, id string) error {
 			if err != nil {
 				return HTTPError(ErrorGettingBuildDependencyStatus)
 			}
+
+			var us *UploadStatus
+			// only single upload target in addition to Koji is allowed
+			if len(buildJobResult.TargetResults) > 2 {
+				return HTTPError(ErrorSeveralUploadTargets)
+			}
```

Comment:
That one of them is actually a koji target is ensured by `enqueueKojiCompose`, i.e. the specialised compose handler we use when `request.Koji` is present. That tripped me up a bit.